merino/web/api_v1.py

```diff
@@ -199,6 +200,44 @@ async def suggest(
         if client_variants
         else [],
     )
+
+    # make sure the request received is for the accuweather provider before pulling out the ttl
+    # and setting it as the cache-control header for the reponse
+    if len(search_from) == 1 and search_from[0].name == "accuweather":
+        accuweather_provider = search_from[0]
+        weather_report_ttl = None
+
+        # at this point we know all the suggestions are accuweather suggestions. We want pull
+        # out the weather_report_ttl from the custom_details property and then remove the
+        # custom_details property since we don't want to send it back as part of the response.
+        for suggestion in suggestions:
+            if hasattr(suggestion, "custom_details") and hasattr(
+                suggestion.custom_details, "weather"
+            ):
+                weather_report_ttl = (
+                    suggestion.custom_details.weather.weather_report_ttl
+                )
+                delattr(suggestion, "custom_details")
+
+        # Build the cache-control response header based on the weather_report_ttl value
+        if weather_report_ttl is not None:
+            response_headers = {
+                "Cache-control": f"public, max-age={weather_report_ttl}"
+            }
+        else:
+            default_weather_report_ttl = (
+                accuweather_provider.backend.cached_default_weather_report_ttl_sec
+            )
+            response_headers = {
+                "Cache-control": f"public, max-age={default_weather_report_ttl}"
```

Comment:
```suggestion
                "Cache-control": f"private, max-age={default_weather_report_ttl}"
```

Let's do private here as the weather suggestion may be customized (for specific locations) for the user.